test/conformance/device/fixtures.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#ifndef UUR_DEVICE_TEST_FIXTURES_H_INCLUDED
+#define UUR_DEVICE_TEST_FIXTURES_H_INCLUDED
+
+#include <uur/fixtures.h>
+
+namespace uur {
+namespace device {
+
+inline std::pair<bool, std::vector<ur_device_handle_t>>
+GetDevices(ur_platform_handle_t platform) {
+  uint32_t count = 0;
+  if (urDeviceGet(platform, UR_DEVICE_TYPE_ALL, 0, nullptr, &count)) {
+    return {false, {}};
+  }
+  if (count == 0) {
+    return {false, {}};
+  }
+  std::vector<ur_device_handle_t> devices(count);
+  if (urDeviceGet(platform, UR_DEVICE_TYPE_ALL, count, devices.data(),
+                  nullptr)) {
+    return {false, {}};
+  }
+  return {true, devices};
+}
+
+inline bool
+hasDevicePartitionSupport(ur_device_handle_t device,
+                          const ur_device_partition_property_flags_t property) {
+  ur_device_partition_property_flags_t flags = 0;
+  auto result = urDeviceGetInfo(device, UR_DEVICE_INFO_PARTITION_PROPERTIES,
+                                sizeof(flags), &flags, nullptr);
+  if (result != UR_RESULT_SUCCESS) {
+    return false;
+  }
+  return (flags & property);
+}
+
+struct urDevicesTest : urPlatformTest {
```

Comment:
The reason I've split it out is because there is another similarly named test {`urDevicesTest`](https://github.com/oneapi-src/unified-runtime/blob/main/test/conformance/testing/uur/fixtures.h#L28) which will be used in parametrized tests in the test suite that use the Devices Platform. 
I'm happy to move it into the shared fixtures header - but maybe it should be called something different. I think urDevicesTest & urDeviceTest might be a bit ambiguous as to their usage?

Reply:
What about calling it `urAllDevicesTest` and moving it to the shared fixtures?

Reply:
👍